### ISSUE TITLE
Prompts are not assigned to images with [filewords]

### ISSUE THREAD

**Issue by gregniuki** [d8ahazard/sd_dreambooth_extension#264]
My images names are made of random numbers and each image has corresponding txt file with prompt .
Should field Class prompt be left empty or with string “ [filewords] “?
my instance prompt is  “photo of xyz” .
Previous version was working without filewords

**Comment by d8ahazard** [comment]
What is in the prompt(s) in the txt files? You can do several things.

If your "Total number of class/reg images..." field is greater than 0, then
you'll want to put something in the class prompt field. If it's 0, then you
can just ignore anything related to class/class images.

If you want to generate class images, then it depends on what's in the
txt prompt files. If the prompt is like "a photo of xyz in a forest", then
I'd set the class prompt and instance prompt fields to [filewords],
"existing prompt contents" to "Instance Token + Description", instance
token to "xyz", and class token to whatever "xyz" is.

You can also make the sample prompt [filewords], and it will use the
prompts in your images as the base for what to generate for sample outputs.

**Comment by gregniuki** [comment]
I have 1000 random images in similar style and 1000 corresponding txt files generated automatically with train/ preproces images/ blip in automatic1111 . It’s trained with batch size 10 and 1000 epoch. Which means I’m training 1000 different classes with its prompts but similar style. It’s trained on 3060 12gb . One iteration takes 12-18 s . But I tested the results and names are mixed up.

**Comment by kernelator** [comment]
I am not sure how this new [filewords] option is supposed to work. I have training images with the following structure:
"[ClassName] [ImageNumber]" as in "dog face 1", "dog face 2", "dog face 3"
I have more than one Class in my folder so some other images have this other file name:
"dog body 1", "dog body 2", "dog body 3"
Overall my folder may contains:
"dog face 1", "dog face 2", "dog face 3", "dog body 1", "dog body 2", "dog body 3"

Given this, dog name would be the Instance Prompt correct? So in the UI, I would input "fido" for the dog's name (assuming fido is a unique token)

For Classification Dataset Directory, then I imagine I would have to leave it blank since I cannot have two different paths for the two different classes.

For Existing Prompt Contents I would select Class Token + Description, right?

For Instance token to swap, I don't think it should matter, as the Instance Prompt was explicitly specified, in this case it seems to me that you could either leave it blank or input "fido" again, but I am not sure if this is the case.

For Class token(s) to swap, can be comma-separated, I am also not to sure how this works, I thought this would be the different classes that may be included in the folder, as in "dog face", "dog body" and that this would be used as a way to separate class, from instance, from descriptor in the file names. If this is the intention then it does not appear to be working correctly as when I see the classification images (regularization) being generated, the prompts are not separating the description (I am assuming in this case the description would be the file number) from the class name, so the prompts for classification are showing up in the following way:
"dog face 80" instead of "dog face", this of course could lead to an older looking dog than intended.

Perhaps I am misunderstanding these new fields, it would be nice if we could get some clarification on how we are meant to use them.

**Comment by d8ahazard** [comment]
So, let's see here.

"fido" or a more unique word for your instance token. "fydoh" is less likely to be an existing thing. "dog" would be your class token. [filewords] for Instance, class, and sample prompts. Class Token + Description, as we don't have "fido" or "fydoh" in any of the filenames.

Just for completeness, if you had "fido dog face 1", "fido dog face 2", then it'd be "Class Token + Instance Token + Description", and if they were "fido face 1", "fido body 1", etc. you'd do "Instance Token + Description".

It's a bit to wrap your head around at first, but the end result is that you will have class images named just like your instance image, and captions for instance and sample images with the subject and class swapped, so "fido dog face" and "fido dog body". based on "dog face 1" and "dog body 10". 

Make sense?

**Comment by kernelator** [comment]
Ok, so it does appear that I got the intention correctly. But then how would I go about avoiding including the description in the prompts for the classification images to be generated pre-training. As I said above I am seeing images with the prompt:
"dog face 80" instead of "dog face", with 80 being the 80th dog face in the file name, which is the description part of the file name, this could lead to an older looking dog than intended.

**Comment by d8ahazard** [comment]
That's a good one. I think I'll have to look at the code for the bit that
parses filenames. My initial inclination is to just filter any non-alpha
characters from the prompts, but then that would lead to issues somewhere
else. I'll give it some thought. For the most part, I don't think the
individual numbers will have too much impact, but I'm totally just guessing.

**Comment by kernelator** [comment]
My guess is that it will increasingly skew everything toward being old the larger then number of images generated is, as there is no upper bound, but lower bound is likely 0 or 1, so if class is person you will get a few babies a lot of old people and a ton of vampires. 

I think it makes sense to use the fields "Instance token to swap" and "Class token(s) to swap" for string search in the file name and assume any value not included in either of these two fields to be the description and thus have it excluded from the classification prompt.

**Comment by gregniuki** [comment]
When I use’[filewords]’ string in Sample image Prompt field it’s just naming each file with [filewords]100.png it means it’s also not using any prompt included in txt file just using word [file words] for each prompt which is error.

**Comment by d8ahazard** [comment]
> When I use’[filewords]’ string in Sample image Prompt field it’s just naming each file with [filewords]100.png it means it’s also not using any prompt included in txt file just using word [file words] for each prompt which is error.

Yes, that's definitely not the intended behavior. I see you opened this issue 14 hours ago. With your most recent comment, were you running the latest version of the extension? I just added [filewords] support for sample prompt generation yesterday.

**Comment by gregniuki** [comment]
Sorry I didn’t updated . Is Batch size affecting trained model ? When I use 10 it’s just mixing 10 prompts with 10 images and it’s resulting in model tokens to be mixed . In the end  no matter what token you apply the result image were the same for the same seed.

**Comment by d8ahazard** [comment]
> Sorry I didn’t updated . Is Batch size affecting trained model ? When I use 10 it’s just mixing 10 prompts with 10 images and it’s resulting in model tokens to be mixed . In the end no matter what token you apply the result image were the same for the same seed.

Can you give an example of what you mean, please?

**Comment by d8ahazard** [comment]
This should be resolved with the latest release. Feel free to re-open if not.
